sys/include/net/ng_slip.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_ng_slip SLIP
+ * @ingroup     net
+ * @brief       Provides a SLIP interface over UART utilizing
+ *              @ref driver_periph_uart.
+ * @see         <a href="https://www.ietf.org/rfc/rfc1055">RFC 1055</a>
+ * @{
+ *
+ * @file
+ * @brief       SLIP interface defintion
+ *
+ * @author      Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+
+#ifndef NG_SLIP_H_
+#define NG_SLIP_H_
+
+#include <inttypes.h>
+
+#include "net/ng_netbase.h"
+#include "periph/uart.h"
+#include "ringbuffer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !UART_NUMOF
+/**
+ * @brief   Guard type for boards that do not implement @ref driver_periph_uart
+ */
+typedef uint8_t uart_t;
+#endif
+
+/**
+ * @brief   Descriptor for the UART interface.
+ */
+typedef struct {
+    uart_t uart;            /**< the UART interface */
+    ringbuffer_t *in_buf;   /**< input buffer */
+    ringbuffer_t *out_buf;  /**< output buffer */
```

Comment:
also include the buffer memory into the file descriptor and initialize the ringbuffers inside the init() function.